ArjunaJTA/jdbc/classes/com/arjuna/ats/internal/jdbc/DirectRecoverableConnection.java

```diff
@@ -246,7 +246,9 @@ public void closeCloseCurrentConnection() throws SQLException
             if (_theConnection != null)
             {
                 _theConnection.close();
-                _theConnection = null;
+				_theConnection = null;
+				_theXAResource = null;
```

Comment:
A call to
```java
reset ()
```
would have been neater here. And similarly for IndirectRecoverableConnection below.